### PR TITLE
Search flight page

### PR DESCRIPTION
**Pull request by AnastasiaBerlina**

### CODE REVIEW

**Comment by AnastasiaBerlina** [comment]
[[BE] Flights page + add necessary methods to Flight](https://trello.com/c/hLtY4ESC/23-be-flights-page-add-necessary-methods-to-flight)

File: src/main/java/com/epam/lowcost/DAO/FlightDAO.java [hunk 1]
```diff
@@ -11,5 +12,6 @@
     Flight addNewFlight(Flight flight);
     Flight deleteFlight(Long id);
     Flight updateFlight(Flight flight);
+    List <Flight> getByFromToDate (String departureAirport, String arrivalAirport, LocalDateTime departureDate,LocalDateTime arrivalDate);
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/FlightDAO.java]: Is this filter really necessary ?

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 2]
```diff
@@ -30,6 +30,17 @@ public String findFlightById(@RequestParam Long id, Model model) {
         return "flights";
     }
 
+
+
+    @GetMapping(value = "/search")
+    public String findFlightByFromToDate(@RequestParam Map<String, String> params, Model model) {
+        model.addAttribute("flights", flightService.getByFromToDate
+                (params.get("departureAirport"), params.get("arrivalAirport"),
+                        LocalDateTime.parse(params.get("departureDate")+"T00:00:00"),
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: What is it ?

File: src/main/java/com/epam/lowcost/util/DateFormatter.java [hunk 3]
```diff
@@ -8,12 +8,18 @@
 @UtilityClass
 public class DateFormatter {
 
-    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss");
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    private final DateTimeFormatter customFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd 00:00:00");
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/util/DateFormatter.java]: Why you use additional format ? Why you don't use one common format ? I think you can deal with it

File: src/main/java/com/epam/lowcost/DAO/FlightDAO.java (hunk 1 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/DAO/FlightDAO.java]: yes, we need it when user wants to find flight exactly for his/her demand

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 2 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: user inputs only date of flight and in our database we store data+time, so it is needed to add time to search in db

File: src/main/java/com/epam/lowcost/util/DateFormatter.java (hunk 3 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/util/DateFormatter.java]: deal with it

File: src/main/java/com/epam/lowcost/DAO/FlightDAO.java (hunk 1 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/DAO/FlightDAO.java]: What do you do in real life when you choose flight? You choose airports(from/to), and range of date only of departure. You never choose range between departure time and arrival time, don't you ?

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 2 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: I said about formatter

File: src/main/java/com/epam/lowcost/DAO/FlightDAO.java (hunk 1 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/DAO/FlightDAO.java]: yes, this is exactly what happens in the code, arrivalDate is only the name of variable

Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/DAO/FlightDAO.java]: corrected

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 2 above)
Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: didn't understand

Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: `params.get("departureDate")+"T00:00:00")`

Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: In the project you should use alone formatter for all dates

Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: I dont use any formatter there, I just add time to the date I get from page
What about formatter - I fixed it

Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Ok, I can ask differently. Why u use concatenation ?

Comment (AnastasiaBerlina) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: Ok, got it. Replaced concatenation with appropriate method

**Pull request merged**